### PR TITLE
Offline state refactor robot status

### PR DESCRIPTION
**Pull request by oysand**

### CODE REVIEW

File: src/isar/state_machine/states/idle.py [hunk 1]
```diff
@@ -37,4 +48,36 @@ def _run(self) -> None:
                 break
             time.sleep(self.state_machine.sleep_time)
 
+            time_from_last_robot_status_poll = (
+                time.time() - self.last_robot_status_poll_time
```
Comment (oysand) [inline src/isar/state_machine/states/idle.py:52]: Reset time

File: src/isar/state_machine/states/idle.py [hunk 2]
```diff
@@ -37,4 +48,38 @@ def _run(self) -> None:
...
+            try:
+                robot_status: RobotStatus = self.robot_status_thread.get_output()
+            except ThreadedRequestNotFinishedError:
+                time.sleep(self.state_machine.sleep_time)
+                continue
+
+            except (RobotException,) as e:
+                self.logger.error(
+                    f"Failed to get robot status because: {e.error_description}"
+                )
```
Comment (andchiind) [inline src/isar/state_machine/states/idle.py:77]: Could we maybe move this to another function in order to shorten the function and make it a bit easier to read?

File: src/isar/state_machine/states/idle.py [hunk 3]
```diff
@@ -37,4 +48,38 @@ def _run(self) -> None:
...
+
+            self.last_robot_status_poll_time = time.time()
+
+            if robot_status == RobotStatus.Offline:
+                transition = self.state_machine.robot_turned_offline  # type: ignore
+                break
+
```
Comment (andchiind) [inline src/isar/state_machine/states/idle.py:84]: Alternatively we could move all this to another function

File: src/isar/state_machine/state_machine.py [hunk 4]
```diff
@@ -580,25 +585,33 @@ def publish_step_status(self, step: Step) -> None:
...
             retain=False,
         )
 
+    def _current_status(self) -> RobotStatus:
+        if self.current_state == States.Idle:
+            return RobotStatus.Available
+        elif self.current_state == States.Offline:
+            return RobotStatus.Offline
+        else:
+            return RobotStatus.Busy
```
Comment (andchiind) [inline src/isar/state_machine/state_machine.py:613]: Do we need a blocked state to map to RobotStatus.Blocked?

Comment (andchiind) [reply src/isar/state_machine/state_machine.py:613]: Not needed in this PR, but might be re-added in other PR

File: src/isar/state_machine/states/idle.py (hunk 2 above)
Comment (andchiind) [reply src/isar/state_machine/states/idle.py:77]: We can standardise this in another PR

File: src/isar/state_machine/states/idle.py (hunk 3 above)
Comment (andchiind) [reply src/isar/state_machine/states/idle.py:84]: We can standardise this in another PR

**Pull request merged**